scan/connectivity.go

```diff
@@ -0,0 +1,80 @@
+package scan
+
+import (
+	"crypto/tls"
+	"errors"
+	"net"
+	"strings"
+)
+
+// Connectivity contains scanners testing basic connectivity to the host
+var Connectivity = &Family{
+	Name:        "Connectivity",
+	Description: "Scans for basic connectivity with the host through DNS and TCP/TLS dials",
+	Scanners: []*Scanner{
+		{
+			"DNSLookup",
+			"Host can be resolved through DNS",
+			dnsLookupScan,
+		},
+		{
+			"TCPDial",
+			"Host accepts TCP connection",
+			tcpDialScan,
+		},
+		{
+			"TLSDial",
+			"Host can perform TLS handshake",
+			tlsDialScan,
+		},
+	},
+}
+
+type lookupAddrs []string
+
+func (addrs lookupAddrs) String() string {
+	return strings.Join(addrs, "\n")
+}
+func (addrs lookupAddrs) Describe() string {
+	return "List of host's addresses returned by DNS lookup"
+}
+
+// dnsLookupScan tests that DNS resolution of the host returns at least one address
+func dnsLookupScan(host string) (grade Grade, output Output, err error) {
+	host, _, err = net.SplitHostPort(host)
+	if err != nil {
+		return
+	}
```

Comment:
It's a style nit, but I find the code is more readable if you put a blank line here (and L53).